.github/workflows/build.yaml

```diff
@@ -1,15 +1,31 @@
 name: Build Docker image
-on: [push]
+on:
+  push:
+  release:
+    types: [created, edited, published]
 jobs:
   push_to_registry:
     name: Push Docker image to GitHub Container Registry
     runs-on: ubuntu-latest
     steps:
       - name: Check out repo
         uses: actions/checkout@v2
-      - name: Push to GitHub Container Registry
-        uses: VaultVulp/gp-docker-action@1.1.7
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          image-name: xnat
-          extract-git-tag: true
+          registry: ghcr.io
+          username: bnc-it
+          password: ${{ secrets.CR_PAT }}
+      - name: Build Docker image
+        run: docker-compose build
+      - name: Push dev image to GitHub Container Registry
+        run: |
+          RELEASE_IMAGE="$(cat docker-compose.yaml | sed -n 's/^[ ]*image: \(ghcr.*$\)/\1/p')"
+          DEV_IMAGE="$(cat docker-compose.yaml | sed -n 's/^[ ]*image: \(ghcr.*\):.*$/\1/p'):${GITHUB_SHA:0:7}"
+          docker tag "${RELEASE_IMAGE}" "${DEV_IMAGE}"
+          docker push "${DEV_IMAGE}"
+      - name: Push release image to GitHub Container Registry
+        if: ${{ github.event_name == 'release' }}
+        run: |
+          RELEASE_IMAGE="$(cat docker-compose.yaml | sed -n 's/^[ ]*image: \(ghcr.*$\)/\1/p')"
```

Comment:
I guess this is where you could use the `set-env` so you don't need to re-define it.

What are the versions being set to?  the version in the docker compose for release, and for the dev you attach the SHA?

Reply:
That's exactly right! On release, it takes whatever version is listed in the `docker-compose.yaml` file (I've been using the `xnat-web` version number there). On push, it builds a dev image using the first 7 characters of the git sha. 7 characters is an old convention from `git log --oneline`.